mmv1/third_party/terraform/tests/resource_storage_bucket_test.go.erb

```diff
@@ -980,6 +981,109 @@ func TestAccStorageBucket_retentionPolicyLocked(t *testing.T) {
 	})
 }
 
+func TestLabelDiffSuppress(t *testing.T) {
+	cases := map[string]struct {
+		K, Old, New        string
+		ExpectDiffSuppress bool
+	}{
+		"missing goog-dataplex-asset-id": {
+			K:                  "labels.goog-dataplex-asset-id",
+			Old:                "test-bucket",
+			New:                "",
+			ExpectDiffSuppress: true,
+		},
+		"explicit missing goog-dataplex-asset-id": {
+			K:                  "labels.goog-dataplex-asset-id",
+			Old:                "test-bucket",
+			New:                "test-bucket",
+			ExpectDiffSuppress: false,
+		},
+		"missing goog-dataplex-lake-id": {
+			K:                  "labels.goog-dataplex-lake-id",
+			Old:                "test-lake",
+			New:                "",
+			ExpectDiffSuppress: true,
+		},
+		"explicit goog-dataplex-lake-id": {
+			K:                  "labels.goog-dataplex-lake-id",
+			Old:                "test-lake",
+			New:                "test-lake",
+			ExpectDiffSuppress: false,
+		},
+		"missing goog-dataplex-project-id": {
+			K:                  "labels.goog-dataplex-project-id",
+			Old:                "test-project-12345",
+			New:                "",
+			ExpectDiffSuppress: true,
+		},
+		"explicit goog-dataplex-project-id": {
+			K:                  "labels.goog-dataplex-project-id",
+			Old:                "test-project-12345",
+			New:                "test-project-12345",
+			ExpectDiffSuppress: false,
+		},
+		"missing goog-dataplex-zone-id": {
+			K:                  "labels.goog-dataplex-zone-id",
+			Old:                "test-zone1",
+			New:                "",
+			ExpectDiffSuppress: true,
+		},
+		"explicit goog-dataplex-zone-id": {
+			K:                  "labels.goog-dataplex-zone-id",
+			Old:                "test-zone1",
+			New:                "test-zone1",
+			ExpectDiffSuppress: false,
+		},
+		"labels.%": {
+			K:                  "labels.%",
+			Old:                "5",
+			New:                "1",
+			ExpectDiffSuppress: true,
+		},
+		"deleted custom key": {
+			K:                  "labels.my-label",
+			Old:                "my-value",
+			New:                "",
+			ExpectDiffSuppress: false,
+		},
+		"added custom key": {
+			K:                  "labels.my-label",
+			Old:                "",
+			New:                "my-value",
+			ExpectDiffSuppress: false,
+		},
+	}
+	for tn, tc := range cases {
+		if testLabelDiffSuppress(tc.K, tc.Old, tc.New) != tc.ExpectDiffSuppress {
+			t.Errorf("bad: %s, %q: %q => %q expect DiffSuppress to return %t", tn, tc.K, tc.Old, tc.New, tc.ExpectDiffSuppress)
```

Comment:
```suggestion
	for tn, tc := range cases {
		if resourceDataplexLabelDiffSuppress(tc.K, tc.Old, tc.New) != tc.ExpectDiffSuppress {
			t.Errorf("bad: %s, %q: %q => %q expect DiffSuppress to return %t", tn, tc.K, tc.Old, tc.New, tc.ExpectDiffSuppress)
```

No need to create a new function. `resourceDataplexLabelDiffSuppress` is the function we want to test and we can just call it directly here.

Reply:
Thanks, @shuyama1 ! I think we need to adjust the function call to `resourceDataplexLabelDiffSuppress(tc.K, tc.Old, tc.New, nil)` instead of `resourceDataplexLabelDiffSuppress(tc.K, tc.Old, tc.New)` as the `resourceDataplexLabelDiffSuppress` expects four arguments and can't miss the last one - even though it is not used in the function.. :(

Reply:
Make sense! Thanks!